syntax = "proto3";

package user.v1;
import "errors/errors.proto";

option go_package = "api/user/service/v1/pb;v1";

enum UserErrorReason {
  option (errors.default_code) = 500;

  UNKNOWN_ERROR = 0;
  VERIFY_PASSWORD_FAILED = 1 [(errors.code) = 401];
  VERIFY_CODE_FAILED = 2 [(errors.code) = 401];
  USERNAME_CONFLICT = 3 [(errors.code) = 409];
  PHONE_CONFLICT = 4 [(errors.code) = 409];
  EMAIL_CONFLICT = 5 [(errors.code) = 409];
  USER_NAME_CONFLICT = 6 [(errors.code) = 409];
  GET_ACCOUNT_FAILED = 7 [(errors.code) = 404];
  GET_PROFILE_FAILED = 8 [(errors.code) = 404];
  GET_PROFILE_UPDATE_FAILED = 9 [(errors.code) = 404];
  GET_ACHIEVEMENT_FAILED = 10 [(errors.code) = 404];
  SEND_CODE_FAILED = 11;
  SET_PHONE_FAILED = 12;
  SET_EMAIL_FAILED = 13;
  SET_PROFILE_FAILED = 14;
  SET_PROFILE_UPDATE_FAILED = 15;
  SET_USERNAME_FAILED = 16;
  SET_IMAGE_FAILED = 17;
  LOGIN_FAILED = 18;
  REGISTER_FAILED = 19;
  RESET_PASSWORD_FAILED = 20;
  PROFILE_REVIEW_MODIFY_FAILED = 21;
  PROFILE_UPDATE_MODIFY_FAILED = 22;
}
